octane/evmengine/keeper/db.go

```diff
@@ -113,3 +113,33 @@ func (k *Keeper) listWithdrawalsByAddress(ctx context.Context, withdrawalAddr co
 
 	return withdrawals, nil
 }
+
+// EligibleWithdrawals returns all withdrawals created below the specified height,
+// sorted by the id (oldest to newest), limited by the provided count.
+func (k *Keeper) EligibleWithdrawals(ctx context.Context) ([]*Withdrawal, error) {
+	height := sdk.UnwrapSDKContext(ctx).BlockHeight()
+	// Note: items are ordered by the id in ascending order (oldest to newest).
+	iter, err := k.withdrawalTable.List(ctx, WithdrawalPrimaryKey{})
+	if err != nil {
+		return nil, errors.Wrap(err, "list withdrawals")
+	}
+	defer iter.Close()
+
+	var withdrawals []*Withdrawal
+	for iter.Next() {
+		val, err := iter.Value()
+		if err != nil {
+			return nil, errors.Wrap(err, "get withdrawal")
+		}
+
+		if val.GetCreatedHeight() < uint64(height) {
+			withdrawals = append(withdrawals, val)
+
+			if uint64(len(withdrawals)) == k.maxWithdrawalsPerBlock {
+				break
+			}
+		}
```

Comment:
You only break out early when limit is reach, I would suggest to also break out if `height` is reached, otherwise you iterate over all withdrawals unnecasrily which increases load.

Also, best to avoid nesting if possible.

Also, can use the `umath.Len` function

```
if val.GetCreatedHeight() >= uint64(height) {
  // Withdrawals created in this block are not eligible
  break
} 

withdrawals = append(withdrawals, val)

if umath.Len(withdrawals) >= k.maxWithdrawalsPerBlock {
  break
}

```